parser/src/aspects/block_parser.rs

```diff
@@ -0,0 +1,194 @@
+use lexer::token::TokenType;
+use crate::aspects::base_parser::BaseParser;
+use crate::ast::Expr;
+use crate::parser::{Parser, ParseResult};
+use crate::ast::statement::Block;
+
+pub trait BlockParser<'a> {
+    fn block(&mut self) -> ParseResult<Expr<'a>>;
+}
+
+impl<'a> BlockParser<'a> for Parser<'a> {
+    fn block(&mut self) -> ParseResult<Expr<'a>> {
+        self.expect_token(TokenType::CurlyLeftBracket, "expected start of block expression")?;
+        let block = |exprs| Ok(Expr::Block(Block {
+            exprs
+        }));
+        let mut expressions: Vec<Expr<'a>> = Vec::new();
+        //if the block is empty
+        if self.meet_token(TokenType::CurlyRightBracket) {
+            return block(expressions);
+        }
+        loop {
+            if self.meet_token(TokenType::CurlyRightBracket) {
+                break;
+            }
+
+            let expression = self.statement()?;
+            expressions.push(expression);
+            if self.meet_token(TokenType::CurlyRightBracket) {
+                break;
+            }
+
+
+            self.expect_token(TokenType::SemiColon, "")
+                .or_else(|_| self.expect_token(TokenType::NewLine, "expected new line or semicolon"))?;
```

Comment:
The call parser also meets the `TokenType::NewLine`, so it should be removed there.

Reply:
Not all expressions handle them, Moreover i think any expressions should not handle their termination tokens (newline, semicolons etc) in order to be more flexible with parent expressions such as parenthesis and blocks (i'd be easier to parse childs i think). 
But i think we should reconsider this once the ParserCursor will be OP as its main goal is to facilitate token manipulation.

Reply:
But you right, as the call parser aspect currently consumes the newline, it'll throw here, gotta fix that tomorrow by avoiding the call aspect to consume it.